src/openrct2/object/WallSceneryEntry.cpp

```diff
@@ -0,0 +1,15 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2024 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#include "WallSceneryEntry.h"
+
+DoorSoundType WallSceneryEntry::getDoorSoundType() const
+{
+    return static_cast<DoorSoundType>((flags2 & WALL_SCENERY_2_DOOR_SOUND_MASK) >> WALL_SCENERY_2_DOOR_SOUND_SHIFT);
+}
```

Comment:
Would it be better to return a `std::optional<DoorSoundType>`? Then we wouldn't need to have `null` entries on the vector.

Reply:
I don't think std::optional makes sense here, its a basic enumeration, this is also a direct mapping from the object data.